pkg/apis/externaldns/types.go

```diff
@@ -61,28 +78,47 @@ func NewConfig() *Config {
-	flags.StringVar(&cfg.Zone, "zone", "", "the ID of the hosted zone to target")
-	flags.StringVar(&cfg.Domain, "domain", "example.org.", "the name of the top-level domain to manage")
-	flags.StringArrayVar(&cfg.Sources, "source", nil, "the sources to gather endpoints: [service, ingress], e.g. --source service --source ingress")
-	flags.StringVar(&cfg.Provider, "provider", "", "the DNS provider to materialize the records in: <aws|google>")
-	flags.StringVar(&cfg.GoogleProject, "google-project", "", "gcloud project to target")
-	flags.StringVar(&cfg.Policy, "policy", "sync", "the policy to use: <sync|upsert-only>")
-	flags.StringVar(&cfg.Compatibility, "compatibility", "", "enable to process annotation semantics from legacy implementations: <mate|molecule>")
-	flags.StringVar(&cfg.MetricsAddress, "metrics-address", defaultMetricsAddress, "address to expose metrics on")
-	flags.StringVar(&cfg.LogFormat, "log-format", defaultLogFormat, "log format output: <text|json>")
-	flags.DurationVar(&cfg.Interval, "interval", time.Minute, "interval between synchronizations")
-	flags.BoolVar(&cfg.Once, "once", false, "run once and exit")
-	flags.BoolVar(&cfg.DryRun, "dry-run", true, "run without updating DNS provider")
-	flags.BoolVar(&cfg.Debug, "debug", false, "debug mode")
-	flags.BoolVar(&cfg.Version, "version", false, "display the version")
-	flags.StringVar(&cfg.Registry, "registry", "noop", "type of registry for ownership: <noop|txt>")
-	flags.StringVar(&cfg.RecordOwnerID, "record-owner-id", "", "id for keeping track of the managed records")
-	flags.StringVar(&cfg.TXTPrefix, "txt-prefix", "", `prefix assigned to DNS name of the associated TXT record; e.g. for --txt-prefix=abc_ [CNAME example.org] <-> [TXT abc_example.org]`)
-	flags.StringVar(&cfg.FqdnTemplate, "fqdn-template", "", `fallback template to generate DNS name if annotation is missing; 
-		e.g. --fqdn-template={{.Name}}-{{.Namespace}}.example.org will use service/ingress name/namespace`)
-	return flags.Parse(args)
+	app := kingpin.New("external-dns", "ExternalDNS synchronizes exposed Kubernetes Services and Ingresses with DNS providers.")
+	app.Version(version)
+	app.DefaultEnvars()
+
+	// Flags related to Kubernetes
+	app.Flag("master", "The Kubernetes API server to connect to; defaults to auto-detection from the environment").Default(defaultConfig.Master).StringVar(&cfg.Master)
+	app.Flag("kubeconfig", "Retrieve target cluster configuration from a Kubernetes configuration file; defaults to auto-detection from the environment").Default(defaultConfig.KubeConfig).StringVar(&cfg.KubeConfig)
+	app.Flag("namespace", "Limit sources of endpoints to a specific namespace; defaults to all").Default(defaultConfig.Namespace).StringVar(&cfg.Namespace)
+
+	// Flags related to processing sources
+	app.Flag("source", "The resource types that are queried for endpoints").EnumsVar(&cfg.Sources, "service", "ingress")
```

Comment:
how does this work ? `--source=service,ingress` or `--source=service -- source=ingress` ?